docs/process/experimental-flags.md

```diff
@@ -0,0 +1,98 @@
+# Dart SDK process for changes behind experimental flags
+
+## Problem statement
+
+The Dart SDK ships via a number of channels:
+
+- Via the [Dart SDK](https://www.dartlang.org/tools/sdk#install)
+- Via the [Flutter SDK](http://flutter.io)
+- Internally at Google via an internal channel
+
+Each of these channels use varying release calendars, and keeping these entirely
+aligned is not practical. Further, a number of developers interested in staying
+current with Dart changes, consume the Dart SDK via our [dev
+channel](https://github.com/dart-lang/sdk/wiki/Branches-and-releases). As a
+result, we should anticipate that any dev channel build has the potential to end
+up as a shipped SDK though some channel. And as a consequence of that, it is
+critical that we keep the quality of our dev channel high AND that we keep it
+consistent wrt. which features are complete and supported. At the same time, we
+need the ability to land incomplete features to facilitate a number of tasks:
+
+- Compose a feature that spans multiple components/areas
+- Automated testing prior to the full completion of the feature
+- Allow partner teams to get a preview of a future feature
+- Allow customers to get a preview of a future feature
+- Etc.
+
+## Solution
+
+To ensure completed & supported features can be differentiated from incomplete
+features in-progress, we will put all in-progress features behind a single set
+of flags. Changes to features behind these flags are not considered breaking
+(even if the feature behind the flag was in a stable SDK), and they are subject
+to removal at any time. For details about breaking changes, see the breaking
+change process.
+
+All new features that meet one of the following criteria must be developed
+behind a flag (when possible):
+
+- All breaking changes
+- All language changes
+
+Further, it is recommended to consider developing behind a flag when:
+
+- Landing larger, user-visible changes which will be in an intermediate state
+  over several weeks and perhaps even releases
+- Making changes with the potential to have significant negative performance
+  impact for several weeks and perhaps even across releases
+
+## Details
+
+### Flag format for CLI-based tools
+
+Flags consist of one or more words, combined using dashes, using all lower-case.
+The single source of truth of these flags shall be a single shared .dart file.
+The tools are expected to offer a framework for querying these flags so that the
+implementation of the tools can easily access new flags.
+
+The flags are passed to CLI-based tools using the `--enable-experiment` flag
+(short-form `-e`). Multiple flags can be passed by using multiple flags, or by
+passing several comma-separated flags. Examples:
+
+```
+dart --enable-experiment super_mixins
+dart -e super-mixins
```

Comment:
I'd prefer if we don't support the short form (`-e`), but it's not a blocking issue for me.

Reply:
@mraleph was strongly requesting the short form; can you comment?

Reply:
I'd still say not to the short form as a required flag. I don't want tests starting with `// SharedOptions=-e something`, that's unreadable.

The VM is free to make a short-hand for their own use, they have plenty already, but it's not the flag that we require them to add, or that we will document as the recommended way to enable experiments.

Reply:
I would prefer we had a short form - but I am okay with not having it.

Reply:
Short form removed